lib/tool_shed/util/hg_util.py

```diff
@@ -44,6 +44,9 @@ def clone_repository(repository_clone_url, repository_file_dir, ctx_rev=None):
     if ctx_rev:
         cmd.extend(['-r', ctx_rev])
     cmd.extend([repository_clone_url, repository_file_dir])
+    # Make sure the destination path actually exists before attempting to clone
+    if not os.path.exists(repository_file_dir):
```

Comment:
Hmm, if I try this with my system installed mercurial it works fine:
```
hg clone https://bitbucket.org/brmtron/mathlibcythontest/ /tmp/blurgas/xscsa
```
If older versions don't support this I guess it doesn't harm to do it your way, but shouldn't we only create the parent dir ?

Reply:
This is only a problem if an admin deleted empty directories under `tool_path` after uninstalling repositories.

Reply:
Thanks, got it.